src/GUI/UI/UIGuilds.cpp

```diff
@@ -141,41 +190,29 @@ void GuildDialog() {
     pCurrentFrameMessageQueue->AddGUIMessage(UIMSG_Escape, 1, 0);
 }
 
-//----- (004BC8D5) --------------------------------------------------------
-void SpellBookGenerator() {  // for GuildDialogs
-    ITEM_TYPE pItemNum;   // esi@1
-    int randomnum;  // esi@7
+void generateSpellBooksForGuild() {
+    BuildingType guildType = buildingTable[window_SpeakInHouse->wData.val - 1].uType;
+
+    // Combined guilds exist only in MM6/MM8 and need to be processed separately
+    assert(guildType >= BuildingType_FireGuild && guildType <= BuildingType_DarkGuild);
+
+    DAMAGE_TYPE schoolType = guildSpellsSchool[guildType];
+    PLAYER_SKILL_MASTERY maxMastery = guildSpellsMastery[HOUSE_ID(window_SpeakInHouse->wData.val)];
+    Segment<ITEM_TYPE> spellsForGuild = spellsOfSchool(schoolType, maxMastery);
 
     for (int i = 0; i < 12; ++i) {
-        // TODO(captainurist): clean up these ITEM_TYPE casts.
-        if (buildingTable[window_SpeakInHouse->wData.val - 1].uType >= BuildingType_FireGuild) {
-            if (buildingTable[window_SpeakInHouse->wData.val - 1].uType <= BuildingType_DarkGuild) {
-                pItemNum = ITEM_TYPE(grng->random(word_4F0F30[window_SpeakInHouse->wData.val - HOUSE_FIRE_GUILD_INITIATE_EMERALD_ISLE]) +
-                    11 * std::to_underlying(buildingTable[window_SpeakInHouse->wData.val - 1].uType) +
-                                     345);
-            } else {
-                if (buildingTable[window_SpeakInHouse->wData.val - 1].uType == BuildingType_ElementalGuild)
-                    randomnum = grng->random(4);
-                else if (buildingTable[window_SpeakInHouse->wData.val - 1].uType == BuildingType_SelfGuild)
-                    randomnum = grng->random(3) + 4;
-                else if (buildingTable[window_SpeakInHouse->wData.val - 1].uType == BuildingType_MirroredPath)
-                    randomnum = grng->random(2) + 7;
-                if (buildingTable[window_SpeakInHouse->wData.val - 1].uType <= BuildingType_MirroredPath)
-                    pItemNum = ITEM_TYPE(grng->random(word_4F0F30[window_SpeakInHouse->wData.val - HOUSE_FIRE_GUILD_INITIATE_EMERALD_ISLE]) +
-                        11 * randomnum +
-                                         400);
```

Comment:
Would it make sense to preserve this code under #ifdef for future reference when we get to mm6?

Reply:
I don't think so. Implementing functionality itself is quite easy. I could do it now and guard with debugbreak but we can't map houses ID to max spell mastery for MM6/8. Array word_4F0F30 that used before could be used for houses ID of other games.

Reply:
OK!